client/src/components/utils/Post/index.js

```diff
@@ -0,0 +1,32 @@
+import React from 'react';
+import { Icon } from 'antd';
+import { Link, BrowserRouter as Router } from 'react-router-dom';
+import './style.css'
+
+export default function Post({ onClick, postTitle, id, postType }) {
+  return (
+    <Router>
+      <Link to={`/post/${id}`}>
+        <div className='post'>
+          <p className='post--title'>
+            {postTitle.length > 30 ? postTitle.substring(0, 31) : postTitle}
```

Comment:
@IsraaSulaiman if this text layout is on just one line (it doesn't break to multiple lines) you could try using text-overflow: ellipsis to truncate the text instead of truncating it yourself in javascript.  There are a few css properties you have to set to make it work, but it adds the "..." to the end of your text and looks nice.  The css properties are:
```
overflow: hidden;
white-space: nowrap;
text-overflow: ellipsis;
```
You can read more here: https://developer.mozilla.org/en-US/docs/Web/CSS/text-overflow